ConsumerApi.Tests.Integration/StepDefinitions/MessagesStepDefinitions.cs

```diff
@@ -0,0 +1,118 @@
+using Backbone.ConsumerApi.Sdk.Authentication;
+using Backbone.ConsumerApi.Sdk.Endpoints.Messages.Types.Requests;
+using Backbone.ConsumerApi.Sdk.Endpoints.Messages.Types.Responses;
+using Backbone.ConsumerApi.Sdk.Endpoints.Relationships.Types.Requests;
+using Backbone.ConsumerApi.Sdk.Endpoints.RelationshipTemplates.Types.Requests;
+using Backbone.ConsumerApi.Tests.Integration.Configuration;
+using Backbone.ConsumerApi.Tests.Integration.Extensions;
+using Backbone.ConsumerApi.Tests.Integration.Support;
+using Backbone.Crypto;
+using Microsoft.Extensions.Options;
+
+namespace Backbone.ConsumerApi.Tests.Integration.StepDefinitions;
+
+[Binding]
+[Scope(Feature = "POST Message")]
+internal class MessagesStepDefinitions
+{
+    private Client _sdk1 = null!;
+    private Client _sdk2 = null!;
+    private ApiResponse<SendMessageResponse>? _sendMessageResponse;
+    private readonly ClientCredentials _clientCredentials;
+    private readonly HttpClient _httpClient;
+
+    public MessagesStepDefinitions(HttpClientFactory factory, IOptions<HttpConfiguration> httpConfiguration)
+    {
+        _httpClient = factory.CreateClient();
+        _clientCredentials = new ClientCredentials(httpConfiguration.Value.ClientCredentials.ClientId, httpConfiguration.Value.ClientCredentials.ClientSecret);
+    }
+
+    [Given("Identities i1 and i2 with an established Relationship")]
+    public async Task GivenIdentitiesI1AndI2WithAnEstablishedRelationship()
+    {
+        _sdk1 = await Client.CreateForNewIdentity(_httpClient, _clientCredentials, Constants.DEVICE_PASSWORD);
+        _sdk2 = await Client.CreateForNewIdentity(_httpClient, _clientCredentials, Constants.DEVICE_PASSWORD);
+
+        var createRelationshipTemplateRequest = new CreateRelationshipTemplateRequest
+        {
+            Content = ConvertibleString.FromUtf8("AAA").BytesRepresentation
+        };
+
+        var relationshipTemplateResponse = await _sdk1.RelationshipTemplates.CreateTemplate(createRelationshipTemplateRequest);
+        relationshipTemplateResponse.Should().BeASuccess();
+
+        var createRelationshipRequest = new CreateRelationshipRequest
+        {
+            RelationshipTemplateId = relationshipTemplateResponse.Result!.Id,
+            Content = ConvertibleString.FromUtf8("AAA").BytesRepresentation
+        };
+
+        var createRelationshipResponse = await _sdk2.Relationships.CreateRelationship(createRelationshipRequest);
+        createRelationshipResponse.Should().BeASuccess();
+
+        var completeRelationshipChangeRequest = new CompleteRelationshipChangeRequest
+        {
+            Content = ConvertibleString.FromUtf8("AAA").BytesRepresentation
+        };
+        var acceptRelationChangeResponse =
+            await _sdk1.Relationships.AcceptChange(createRelationshipResponse.Result!.Id, createRelationshipResponse.Result.Changes.First().Id, completeRelationshipChangeRequest);
+        acceptRelationChangeResponse.Should().BeASuccess();
```

Comment:
We should already start trying to optimize this. I'd suggest a class for utility methods in which you have a method like `EstablishRelationshipBetween(Client client1, Client client2`)`.